### PR TITLE
Feature/handle teams

### PR DESCRIPTION
**Pull request by MikolajWojewoda**

### CODE REVIEW

File: src/FantasyFun.API/Controllers/PlayersController.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        {
+            _dbContext = dbContext;
+            _defaultGameTime = gameSettings.DefaultGameTime;
+        }
+
+        [HttpGet]
+        [Route("overall/{overall}")]
+        public async Task<ActionResult<List<string>>> GetAnyPlayers(long overall)
+        {
+            //var db = new FootballDbContext();
```
Comment (ficmichal) [inline src/FantasyFun.API/Controllers/PlayersController.cs]: Please, delete commented code.

File: src/FantasyFun.API/Controllers/PlayersController.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+            var anyPlayers = await _dbContext.PlayerAttributes
+                .Where(l => l.Date <= _defaultGameTime)
+                .OrderByDescending(l => l.Date)
+                .GroupBy(l => l.Player.Name)//.Max(l => l.Select(k => k.Date))
```
Comment (ficmichal) [inline src/FantasyFun.API/Controllers/PlayersController.cs]: Please, delete commented code.

File: src/FantasyFun.API/Controllers/PlayersController.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                .Select(l => new PlayerType(l.FirstOrDefault().Player.Name, l.FirstOrDefault().OverallRating, l.FirstOrDefault().Date))
+                .Take(500)
+                .ToListAsync();
+
+            var anyPlayersWithOverall = anyPlayers
+                .Where(l => l.OverallRating == overall)
+                .Select(l => l.Name)
+                .Take(10)
```
Comment (ficmichal) [inline src/FantasyFun.API/Controllers/PlayersController.cs]: You can extract 10 as new configuration variable. Please, add new property to GameSettings, name it 'PlayersByOverallMaxNumber').

File: src/FantasyFun.API/Controllers/PlayersController.cs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        public async Task<ActionResult<List<string>>> GetAnyPlayers(long overall)
+        {
+            //var db = new FootballDbContext();
+
+            var anyPlayers = await _dbContext.PlayerAttributes
+                .Where(l => l.Date <= _defaultGameTime)
+                .OrderByDescending(l => l.Date)
+                .GroupBy(l => l.Player.Name)//.Max(l => l.Select(k => k.Date))
+                .Select(l => new PlayerType(l.FirstOrDefault().Player.Name, l.FirstOrDefault().OverallRating, l.FirstOrDefault().Date))
+                .Take(500)
```
Comment (ficmichal) [inline src/FantasyFun.API/Controllers/PlayersController.cs:38]: You can make 500 as private variable inside this class. You can name it 'InternalPlayersByOverallMaxNumber'.

File: src/FantasyFun.API/Controllers/TeamsController.cs [hunk 5]
```diff
@@ -2,25 +2,38 @@
...
+        {
+            _dbContext = dbcontext;
+        }
+        
         [HttpGet]
         [Route("{id}")]
         public async Task<ActionResult<TeamName>> GetTeamById(long id)
         {
-            var db = new FootballDbContext();
+            //var db = new FootballDbContext(new DbSettings());
```
Comment (ficmichal) [inline src/FantasyFun.API/Controllers/TeamsController.cs]: Please remove commented code.

File: src/FantasyFun.API/Program.cs [hunk 6]
```diff
@@ -6,14 +9,24 @@ public static void Main(string[] args)
...
+            builder.Services.AddScoped<GameSettings>(serviceProvider => GameSettings);
+
             var app = builder.Build();
 
             app.UseRouting();
 
             app.MapControllers();
+
             app.Run();
+
```
Comment (ficmichal) [inline src/FantasyFun.API/Program.cs]: Please remove blank line.

File: src/FantasyFun.API/Program.cs [hunk 7]
```diff
@@ -6,14 +9,24 @@ public static void Main(string[] args)
...
             var builder = WebApplication.CreateBuilder(args);
 
+            builder.Services.AddRouting();
+
             builder.Services.AddControllers();
 
+            var dbSettings = builder.Configuration.GetSection("Db").Get<DbSettings>();
+            builder.Services.AddScoped<FootballDbContext>(serviceProvider => new FootballDbContext(dbSettings));
+
+            var GameSettings = builder.Configuration.GetSection("GameSettings").Get<GameSettings>();
```
Comment (ficmichal) [inline src/FantasyFun.API/Program.cs]: Let's keep naming convention - camelCase local variables.

File: src/FantasyFun.API/Repositories/FootballDbContext.cs [hunk 8]
```diff
@@ -1,17 +1,18 @@
 ﻿using FantasyFun.API.Models;
 using Microsoft.EntityFrameworkCore;
+using FantasyFun.API.Settings;
 
 namespace FantasyFun.API.Repositories
 {
     public class FootballDbContext : DbContext
     {
-        public const string DbPath = "data\\database.sqlite";
+        //public const string DbPath = "data\\database.sqlite";
```
Comment (ficmichal) [inline src/FantasyFun.API/Repositories/FootballDbContext.cs]: Please remove commented code

File: src/FantasyFun.API/ViewModel/TeamName.cs [hunk 9]
```diff
@@ -4,11 +4,18 @@ public class TeamName
...
         public string ShortName { get; }
+        public long Overall { get; }
 
         public TeamName(string longName, string shortName)
         {
             LongName = longName;
             ShortName = shortName;
         }
+
+        public TeamName(string LongName, long overall)
```
Comment (ficmichal) [inline src/FantasyFun.API/ViewModel/TeamName.cs]: Constructor parematers should be camelCase.

File: src/FantasyFun.API/ViewModel/TeamName.cs [hunk 10]
```diff
@@ -4,11 +4,18 @@ public class TeamName
     {
         public string LongName { get; }
         public string ShortName { get; }
+        public long Overall { get; }
```
Comment (ficmichal) [inline src/FantasyFun.API/ViewModel/TeamName.cs]: Do we need that overall for the team?

**Review by ficmichal — CHANGES_REQUESTED** [review]
Funcionality works fine. Please, fix some minor changes.

File: src/FantasyFun.API/ViewModel/TeamName.cs (hunk 10 above)
Comment (MikolajWojewoda) [reply src/FantasyFun.API/ViewModel/TeamName.cs]: We don't need it. That property and the second constructor in the "TeamName" class appeared just for test. I will delete them.

**Review by ficmichal — APPROVED** [review]

**Pull request merged**